worlds/witness/hints.py

```diff
@@ -386,22 +387,22 @@ def get_hintable_areas(world: "WitnessWorld") -> Tuple[Dict[str, List[Location]]
     return locations_per_area, items_per_area
 
 
-def word_area_hint(world: "WitnessWorld", hinted_area: str, corresponding_items: List[Item]) -> Tuple[str, int]:
+def word_area_hint(world: "WitnessWorld", hinted_area: str, area_items: List[Item]) -> Tuple[str, int, Optional[int]]:
     """
     Word the hint for an area using natural sounding language.
     This takes into account how much progression there is, how much of it is local/non-local, and whether there are
     any local lasers to be found in this area.
     """
 
-    local_progression = sum(item.player == world.player and item.advancement for item in corresponding_items)
-    non_local_progression = sum(item.player != world.player and item.advancement for item in corresponding_items)
+    local_progression = sum(item.player == world.player and item.advancement for item in area_items)
```

Comment:
I suspect you could get slightly better performance by turning this into a loop, to iterate only once and check the advancement condition less
```py
local_progression = 0
non_local_progression = 0
for item in area_items:
    if not item.advancement:
        continue
    if item.player == world.player:
        local_progression += 1
    else:
        non_local_progression += 1
```

But also, I don't personally micro optimizations like that are relevant, and it makes me incomfortable that I am apparently at the stage that I even thought of it. So, if you want, you can do us a both a favor and ignore this comment :)

Reply:
LMAO Yeah that makes sense. I do like optimised code but I do also like short code... So as long as your code is either of those, and you don't write unoptimised long code :D